src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -298,31 +298,36 @@ public static void CreateDirectory(string fullPath)
             {
                 return; // Path already exists and it's a directory.
             }
-            else if (errorInfo.Error == Interop.Error.ENOENT)
+            else if (errorInfo.Error == Interop.Error.ENOENT) // Some parts of the path don't exist yet.
             {
-                // Some parts of the path don't exist yet.
-                CreateParentsAndDirectory(fullPath);
+                // Try create parents bottom to top and track those that could not
+                // be created due to missing parents. Then create them top to bottom.
```

Comment:
I'd keep the comment inside the `CreateParentsAndDirectory` method, as it explains what the method implementation is doing.

Can't you put the `ValueListBuilder<int>` in the method? And use: `using`?

Reply:
Great point! #resolved